### PR TITLE
resolves #32 assign stat.size for files and symbolic links when reading archive

### LINKED ISSUE #32: src() should assign uncompressed size to stat.size for files and symbolic links
When reading an archive, gulp-vinyl-zip has access to the uncompressed size of the entry for files and symbolic links, but does not pass along this information on when creating the stat object for the file. It would be helpful to programs that use gulp-vinyl-zip to extract archives files, and need to know the file size, to be able to access this information without having to resolve the stream to find out.

For symlink, the stat.size should be the length of the path to which it points, which exactly matches the uncompressed size reported by the archive entry.

For other non-file entry types (including directories), the size doesn't have any meaning, so there is no need to assign it.

### PR DESCRIPTION
**Pull request by mojavelinux**
resolves #32

### CODE REVIEW

**Comment by mojavelinux** [comment]
In addition to assigning the stat.size and verifying it in the test, I strengthened the assertions for the stat object.

I also remove an unnecessary assignment of the contents property in the case of a directory. I can move that to a separate PR if you'd like.

File: lib/src/index.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ function toStream(zip) {
 				q.start();
 			}
 		} else if (stat.isSymbolicLink()) {
+			stat.size = entry.uncompressedSize
```
Comment (joaomoreno) [inline lib/src/index.js]:
```suggestion
			stat.size = entry.uncompressedSize;
```

File: lib/src/index.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ function toStream(zip) {
 		};
 
 		if (stat.isFile()) {
-			if (entry.uncompressedSize === 0) {
+			if ((stat.size = entry.uncompressedSize) === 0) {
```
Comment (joaomoreno) [inline lib/src/index.js]:
```suggestion
			stat.size = entry.uncompressedSize;
			if (entry.uncompressedSize === 0) {
```

**Review by joaomoreno — APPROVED** [review]

**Comment by joaomoreno** [comment]
Thanks!

**Pull request merged**